sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -78,29 +82,51 @@ def _start_reader_threads(self):
       t.start()
       self.reader_threads.append(t)
 
+  def _get_source_position(self, range_tracker=None, reader=None):
+    if reader:
+      return reader.get_progress().position.byte_offset
```

Comment:
<!--new_thread; commit:6e2c396b4cdcce20867c583cba65c8664ff79996; resolved:0-->
What about side input sources that do not have byte offsets ?

Reply:
I need Charles or Robert to input here. In my understanding, side inputs always use Avro files - so they are always byte-offset-based sources. Would this be correct? @charlesccychen

Reply:
I think they might now, but we should verify in the case one uses the result of a read (e.g. Create or ReadTextIO) directly as a side input.

Even if it is the case, best to assert this assumption explicitly somewhere.